### PR TITLE
#12: Add user guide

### LINKED ISSUE #12: Add user documentation & Lua comments
- [ ] User guide
- [ ] Lua comments
- [ ] List of dependencies: https://github.com/exasol/exasol-driver-lua/issues/54
- [ ] List of features

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #12

### CODE REVIEW

**Comment by jakobbraun** [comment]
Mention ticket?

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -1,3 +1,239 @@
 # exasol-driver-lua &mdash; User Guide
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:1]:
```suggestion
# exasol-driver-lua – User Guide
```

Comment (jakobbraun) [reply doc/user_guide/user_guide.md:1]: It's more readable in the editor

File: doc/user_guide/user_guide.md [hunk 2]
```diff
@@ -1,3 +1,239 @@
...
 
-User guide will be written in #12.
\ No newline at end of file
+The Exasol Driver for Lua allows you to execute queries on an Exasol database and retrieve the results. This user guide shows how to use the driver.
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:3]:
```suggestion
The Exasol Driver for Lua allows you to execute queries on an Exasol database and retrieve the results. This user-guide shows how to use the driver.
```

File: doc/user_guide/user_guide.md [hunk 3]
```diff
@@ -1,3 +1,239 @@
...
+
+## Installing the Driver and Dependencies
+
+### Install Runtime Dependencies
+
+This project has the following prerequisites:
+
+* [Lua](https://www.lua.org/) &ge; 5.4
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:11]:
```suggestion
* [Lua](https://www.lua.org/) ≥ 5.4
```

File: doc/user_guide/user_guide.md [hunk 4]
```diff
@@ -1,3 +1,239 @@
...
+* [Luarocks](https://luarocks.org/) &ge; 3.8: Package manager for Lua
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:12]:
```suggestion
* [Luarocks](https://luarocks.org/) ≥ 3.8: Package manager for Lua
```

File: doc/user_guide/user_guide.md [hunk 5]
```diff
@@ -1,3 +1,239 @@
...
+* [OpenSSL](https://www.openssl.org/) &ge; 1.1: Library used for TLS connections and RSA encryption
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:13]:
```suggestion
* [OpenSSL](https://www.openssl.org/) ≥ 1.1: Library used for TLS connections and RSA encryption
```

File: doc/user_guide/user_guide.md [hunk 6]
```diff
@@ -1,3 +1,239 @@
...
+
+1. The connection object if the connection was established successfully or `nil` in case of an error
+2. An error in case the operation failed.
+
+#### Connection Properties
+
+When creating a new connection you can specify the following properties:
+
+* `tls_verify` specifies how the database's TLS certificate should be verified. Possible values are:
+  * `none` (default)
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:142]: is this a good default?

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -63,6 +63,15 @@ jobs:
             --deactivate-database-setup
         working-directory: integration-test-docker-environment
 
+      - name: Run example code
+        run: |
+          source $GITHUB_WORKSPACE/integration-test-docker-environment/.build_output/cache/environments/test/environment_info.sh
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply .github/workflows/build.yml:68]: I need to reed that file containing env variables with the DB connection details.
Not sure if `source`ing this in one `run` command also makes the env variables available in the next `run`.

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -63,6 +63,15 @@ jobs:
...
+          eval $(luarocks path)
+          EXASOL_HOST=$ENVIRONMENT_DATABASE_HOST \
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply .github/workflows/build.yml:70]: I don't want to add a Java build just for running the tests. I updated the developer guide and described how to start the docker container.

File: doc/user_guide/user_guide.md (hunk 6 above)
Comment (kaklakariada) [reply doc/user_guide/user_guide.md:142]: No. I created https://github.com/exasol/exasol-driver-lua/issues/53 to investigate this.

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**